templates/vault/0/rancher-compose.yml

```diff
@@ -0,0 +1,67 @@
+version: '2'
+catalog:
+  name: Vault
+  version: 1.0-test-rancher1
+  description: |
+    Installs a Vault container, internal (Rancher) load-balancer, and allows connection to a Consul cluster
+  uuid: vault-rac-0
+  questions:
+    - variable: VAULT_LOCAL_CONFIG
+      label: Provide a default config?
+      description: |
+        This is the embedded JSON format config file that vault will use at startup
+      type: multiline
+      default: |
+        {"backend":{"consul":{"address":"http://<listenAddress>:<listenPort>", "advertise_addr":"http://<advertiseAddress>","path":"<Path>"}}, "listener":{"tcp":{"address":"0.0.0.0:18200","tls_disable":1}}}
+    - variable: useConsul
+      label: Would you like to back Vault with Consul?
+      description: |
+        Selecting yes or no allows you to back your vault repoository with a pre-installed
+        Consul cluster/server
+      default: false
+      type: boolean
+    - variable: consulService
+      label: Specify the Consul service to connect to
+      description: |
+        The pre-installed Consul server to back to
+      default: "select"
+      type: service
+    - variable: volumeDriver
+      label: Driver for volumes
+      description: |
+        How/Where to store your vault config and logs
+      default: local
+      type: enum
+      options:
+        - local
+        - rancher-nfs
+        - rancher-efs
+        - rancher-ebs
+services:
+  vault-lb:
+    scale: 1
+    start_on_create: true
+    lb_config:
+      certs: []
+      port_rules:
+      - priority: 1
+        protocol: tcp
+        service: vault
+        source_port: 8200
+        target_port: 18200

```

Comment:
What do you think if you add another question at catalog, to be able to choose `source_port` number to publish??

Is `target_port` definition right?? What happens if someone edit `VAULT_LOCAL_CONFIG` `"listener"` field??  Have you consider to use `VAULT_REDIRECT_ADDR` env variable as showed at  [vault image docker-entrypoint.sh](https://github.com/hashicorp/docker-vault/blob/75178223957b123d016181e628b0fe6443ab3d49/0.X/docker-entrypoint.sh)?? 

You could ask for a port to listen `VAULT_LISTEN_PORT`, remove `"listener"` field from `VAULT_LOCAL_CONFIG`  and add variable into vault services docker definition
```
+      VAULT_REDIRECT_ADDR: http://0.0.0.0:${VAULT_LISTEN_PORT}
```

and in this file, 
```
+        source_port: ${VAULT_LISTEN_PORT}
+        target_port: ${VAULT_LISTEN_PORT}
```

Reply:
This is a difficult one... the VAULT_REDIRECT_ADDR and VAULT_CLUSTER_ADDR variables to the entrypoint.sh file expect either a fully populated URL.  When passing the "default" value of http://0.0.0.0:${VAULT_LISTEN_PORT}, I get an error back on Vault container startup like so:

```
4/9/2018 11:40:35 AM==> Vault server configuration:
4/9/2018 11:40:35 AM
4/9/2018 11:40:35 AM                     Cgo: disabled
4/9/2018 11:40:35 AM         Cluster Address: https://0.0.0.0:8201
4/9/2018 11:40:35 AM               Log Level: info
4/9/2018 11:40:35 AM                   Mlock: supported: true, enabled: true
4/9/2018 11:40:35 AM        Redirect Address: http://0.0.0.0:8200
4/9/2018 11:40:35 AM                 Storage: consul (HA available)
4/9/2018 11:40:35 AM                 Version: Vault v0.9.6
4/9/2018 11:40:35 AM             Version Sha: 7e1fbde40afee241f81ef08700e7987d86fc7242
4/9/2018 11:40:35 AM
4/9/2018 11:40:35 AM==> Vault server started! Log data will stream in below:
4/9/2018 11:40:35 AM
```
(notice the default IP addresses for cluster and redirect addresses)  followed by...   

```
4/9/2018 11:29:07 AM2018/04/09 18:29:07.839707 [WARN ] physical/consul: reconcile unable to talk with Consul backend: error=service registration failed: Unexpected response code: 400 (Invalid service address)
4/9/2018 11:29:08 AM2018/04/09 18:29:08.603718 [WARN ] physical/consul: check unable to talk with Consul backend: error=Unexpected response code: 500 (CheckID "vault:0.0.0.0:8200:vault-sealed-check" does not have associated TTL)
```
...repeated a few thousand times.  

If, however, I leave both VAULT_REDIRECT_ADDR and VAULT_CLUSTER_ADDR UNSET  the entrypoint script detects the container's IP address and sets the VAULT_REDIRECT_ADDR & VAULT_CLUSTER_ADDR properly (except the hard-coded port of 8200) -or- set if I set them to valid container IP addresses and pass in these manually set values, the container starts up perfectly.

```
4/9/2018 11:32:09 AM==> Vault server configuration:
4/9/2018 11:32:09 AM
4/9/2018 11:32:09 AM                     Cgo: disabled
4/9/2018 11:32:09 AM         Cluster Address: https://172.27.54.61:8201
4/9/2018 11:32:09 AM               Log Level: info
4/9/2018 11:32:09 AM                   Mlock: supported: true, enabled: true
4/9/2018 11:32:09 AM        Redirect Address: http://172.27.54.61:8200
4/9/2018 11:32:09 AM                 Storage: consul (HA available)
4/9/2018 11:32:09 AM                 Version: Vault v0.9.6
4/9/2018 11:32:09 AM             Version Sha: 7e1fbde40afee241f81ef08700e7987d86fc7242
4/9/2018 11:32:09 AM
4/9/2018 11:32:09 AM==> Vault server started! Log data will stream in below:
4/9/2018 11:32:09 AM
```
I may have to extend from the official container to pass the PORT value in, so the entrypoint.sh script can detect the IP address like it does, but use the environment's PORT values.

Reply:
I should have tested more before I commented...  it turns out that the listen port in Vault is hard-coded within the container to be port 8200.  So, regardless of what port you want to listen on, it HAS to go to port 8200 on the Vault container (i.e., your LB can listen on whatever port it wants, as long as it points to port 8200 on the Vault container (which doesn't need to be exposed publicly because the LB handles it).  So, I've reversed some of the changes and re-added the hard 8200 value between the LB and the Vault container.